FakeServer.Test/FakeServerSpecs.cs

```diff
@@ -221,586 +185,487 @@ public async Task GetUsers_PageDoesNotExist_ReturnsEmptyList()
         }
 
         [Fact]
         public async Task Patch_SingleItem()
         {
-            using (var client = new HttpClient())
-            {
-                var collection = "configuration_for_patch";
-                var newConfig = new { url = "192.168.0.1" };
-
-                var content = new StringContent(JsonConvert.SerializeObject(newConfig), Encoding.UTF8, "application/json");
-                var result = await client.PatchAsync($"{_fixture.BaseUrl}/api/{collection}", content);
-                Assert.Equal(HttpStatusCode.NoContent, result.StatusCode);
-
-                result = await client.GetAsync($"{_fixture.BaseUrl}/api/{collection}");
-                result.EnsureSuccessStatusCode();
-                var item = JsonConvert.DeserializeObject<JObject>(await result.Content.ReadAsStringAsync());
-                Assert.Equal(newConfig.url, item["url"].Value<string>());
-                Assert.Equal("abba", item["password"].Value<string>());
-            }
+            var collection = "configuration_for_patch";
+            var newConfig = new { url = "192.168.0.1" };
+
+            var content = new StringContent(JsonConvert.SerializeObject(newConfig), Encoding.UTF8, "application/json");
+            var result = await _fixture.Client.PatchAsync($"api/{collection}", content);
+            Assert.Equal(HttpStatusCode.NoContent, result.StatusCode);
+
+            result = await _fixture.Client.GetAsync($"api/{collection}");
+            result.EnsureSuccessStatusCode();
+            var item = JsonConvert.DeserializeObject<JObject>(await result.Content.ReadAsStringAsync());
+            Assert.Equal(newConfig.url, item["url"].Value<string>());
+            Assert.Equal("abba", item["password"].Value<string>());
         }
 
         [Fact]
         public async Task Delete_SingleItem()
         {
-            using (var client = new HttpClient())
-            {
-                var collection = "configuration_for_delete";
-                var newConfig = new { value = "Something new" };
+            var collection = "configuration_for_delete";
+            var newConfig = new { value = "Something new" };
 
-                var result = await client.DeleteAsync($"{_fixture.BaseUrl}/api/{collection}");
-                Assert.Equal(HttpStatusCode.NoContent, result.StatusCode);
+            var result = await _fixture.Client.DeleteAsync($"api/{collection}");
+            Assert.Equal(HttpStatusCode.NoContent, result.StatusCode);
 
-                result = await client.GetAsync($"{_fixture.BaseUrl}/api/{collection}");
-                Assert.Equal(HttpStatusCode.NotFound, result.StatusCode);
-            }
+            result = await _fixture.Client.GetAsync($"api/{collection}");
+            Assert.Equal(HttpStatusCode.NotFound, result.StatusCode);
         }
 
         [Fact]
         public async Task WebSockets_CalledTwice()
         {
             var are = new AutoResetEvent(false);
```

Comment:
AutoResetEvent is not needed anymore with this implementation. 

Just use
```cs
await WebSocketReceiveHandler();
```
and you can take remove AutoResetEvent.

Reply:
Done.